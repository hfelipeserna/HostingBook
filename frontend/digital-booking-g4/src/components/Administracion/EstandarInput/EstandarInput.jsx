import styles from "./EstandarInput.module.css";

export default function EstandarInput({
    label,
    name,
    onChangeItem,
    setOnChangeItem,
    placeholder,
}) {
    return (
        <div className={styles.containerEstandarInput}>
            <label className={styles.labelEstandarInput}>{label}</label>
            <div className={styles.sombraConAutocompletado} >
                <input
                    type="text"
                    className={`${styles.estandarInput}`}
                    onChange={(e) => {
                        setOnChangeItem(e.target.value)
                    }}
                    value={onChangeItem}
                    name={name}
                    placeholder={placeholder}
                    required
                />
            </div>
        </div>
    )
}
