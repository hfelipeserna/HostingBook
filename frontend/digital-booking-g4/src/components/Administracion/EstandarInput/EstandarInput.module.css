.containerEstandarInput{
    width: 100%;
    display: flex;
    flex-direction: column;
    gap: 0;
}
.sombraConAutocompletado{
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
  width: 100%;
  border-radius: 5px;
  margin-top: 5px;
}

.labelEstandarInput{
    color: var(--color-2);
    font-weight: 700;
    margin-top: 20px;
    cursor: pointer;
}

.estandarInput{
    background-color: var(--color-5);
    color: var(--color-2);
    font-size: 1rem;
    padding: 1vh 40px 1vh 15px;
    width: 100%;
}

.estandarInput:disabled {
  color: rgb(70, 70, 70);
  background-color: rgba(128, 128, 128, 0.25);
  border: 3px solid #DFDFDF;
}


@media (max-width: 480px) {
  .labelEstandarInput{
    margin-top: 7px;
  }

  .estandarInput {
    padding: 0.8vh 40px 0.8vh 15px;
  }
}
  