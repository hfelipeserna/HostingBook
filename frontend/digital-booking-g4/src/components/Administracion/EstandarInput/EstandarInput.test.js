import EstandarInput from "./EstandarInput"
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react"
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import { configure, mount } from "enzyme";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Estandar Input Room test", function () {
    const label = "Titulo"
    const name = "Nombre"
    let onChangeItem = ""
    const setOnChangeItem = (word) => {
        onChangeItem = word
    }

    test("Renderiza Componente Por default", () => {
        const renderizacionSimple = render(<EstandarInput label={label} nombre={name}></EstandarInput>)
        expect(screen.getByText("Titulo"))
        expect(renderizacionSimple).toMatchSnapshot()
    });
    test("Cambia el valor apartir del input", () => {
        const renderizacionCompleta = mount(<EstandarInput setOnChangeItem={setOnChangeItem} label={label} nombre={name} onChangeItem={onChangeItem} ></EstandarInput>)
        const input = renderizacionCompleta.find('input')
        input.simulate('focus');
        input.simulate('change', { target: { value: 'a' } })
        input.simulate('keyDown', {
            which: 27,
            target: {
                blur() {
                    input.simulate('blur');
                },
            },
        })
        expect(renderizacionCompleta).toMatchSnapshot()
        expect(onChangeItem).toEqual("a")
    })
});

