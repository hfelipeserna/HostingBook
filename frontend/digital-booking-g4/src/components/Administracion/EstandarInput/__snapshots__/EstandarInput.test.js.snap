// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Estandar Input Room test Cambia el valor apartir del input 1`] = `
<div
  className="containerEstandarInput"
>
  <label
    className="labelEstandarInput"
  >
    Titulo
  </label>
  <div
    className="sombraConAutocompletado"
  >
    <input
      className="estandarInput"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Estandar Input Room test Renderiza Componente Por default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="containerEstandarInput"
      >
        <label
          class="labelEstandarInput"
        >
          Titulo
        </label>
        <div
          class="sombraConAutocompletado"
        >
          <input
            class="estandarInput"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="containerEstandarInput"
    >
      <label
        class="labelEstandarInput"
      >
        Titulo
      </label>
      <div
        class="sombraConAutocompletado"
      >
        <input
          class="estandarInput"
          required=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
